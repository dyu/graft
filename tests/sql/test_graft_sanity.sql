-- Load the graft vfs and open a new connection using it
.load libgraft
.open file:GontkHa6QVLHYnkyk16wUP?vfs=graft

.databases
.vfsinfo
pragma graft_status;

CREATE TABLE t1(a, b);
INSERT INTO t1 VALUES(1, 2);
INSERT INTO t1 VALUES(3, 4);
SELECT * FROM t1;

BEGIN;
SELECT * FROM t1;
INSERT INTO t1 VALUES(3, 4);
SELECT * FROM t1;
COMMIT;

pragma graft_status;

vacuum;
drop table t1;
vacuum;

select * from dbstat;
