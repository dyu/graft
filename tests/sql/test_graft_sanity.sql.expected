-- Load the graft vfs and open a new connection using it
.load libgraft
.open file:GontkHa6QVLHYnkyk16wUP?vfs=graft

.databases
main: GontkHa6QVLHYnkyk16wUP r/w
.vfsinfo
vfs.zName      = "graft"
vfs.iVersion   = 3
vfs.szOsFile   = 192
vfs.mxPathname = 512
pragma graft_status;
+------------------------+
|      Graft Status      |
+------------------------+
| Graft Status           |
| Current snapshot: None |
| Autosync: false        |
+------------------------+

CREATE TABLE t1(a, b);
INSERT INTO t1 VALUES(1, 2);
INSERT INTO t1 VALUES(3, 4);
SELECT * FROM t1;
+---+---+
| a | b |
+---+---+
| 1 | 2 |
| 3 | 4 |
+---+---+

BEGIN;
SELECT * FROM t1;
+---+---+
| a | b |
+---+---+
| 1 | 2 |
| 3 | 4 |
+---+---+
INSERT INTO t1 VALUES(3, 4);
SELECT * FROM t1;
+---+---+
| a | b |
+---+---+
| 1 | 2 |
| 3 | 4 |
| 3 | 4 |
+---+---+
COMMIT;

pragma graft_status;
+---------------------------------+
|          Graft Status           |
+---------------------------------+
| Graft Status                    |
| Current snapshot: Snapshot[4;2] |
| Autosync: false                 |
+---------------------------------+

vacuum;
drop table t1;
vacuum;

select * from dbstat;
+---------------+------+--------+----------+-------+---------+--------+------------+----------+--------+
|     name      | path | pageno | pagetype | ncell | payload | unused | mx_payload | pgoffset | pgsize |
+---------------+------+--------+----------+-------+---------+--------+------------+----------+--------+
| sqlite_schema | /    | 1      | leaf     | 0     | 0       | 3988   | 0          | 0        | 4096   |
+---------------+------+--------+----------+-------+---------+--------+------------+----------+--------+

SQLite Exit Code = 0
